GANDLF/metrics/classification.py

```diff
@@ -62,46 +64,27 @@ def overall_stats(prediction: torch.Tensor, target: torch.Tensor, params: dict)
                 num_classes=params["model"]["num_classes"],
                 average=average_type_key,
             ),
-            # "aucroc": tm.AUROC(
-            #     task=task,
-            #     num_classes=params["model"]["num_classes"],
-            #     average=average_type_key if average_type_key != "micro" else "macro",
-            # ),
+            "auroc": tm.AUROC(
+                task=task,
+                num_classes=params["model"]["num_classes"],
+                average=average_type_key if average_type_key != "micro" else "macro",
+            ),
         }
         for metric_name, calculator in calculators.items():
-            if metric_name == "aucroc":
-                one_hot_preds = one_hot(
-                    prediction.long(), num_classes=params["model"]["num_classes"]
-                )
+            # TODO: AUROC needs to be properly debugged for multi-class problems - https://github.com/mlcommons/GaNDLF/issues/817
+            if metric_name == "auroc" and params["model"]["num_classes"] == 2:
                 output_metrics[metric_name] = get_output_from_calculator(
-                    one_hot_preds.float(), target, calculator
+                    prediction, target, calculator
                 )
-            else:
+            elif metric_name != "auroc":
                 output_metrics[metric_name] = get_output_from_calculator(
                     prediction, target, calculator
                 )
 
-    #### HERE WE NEED TO MODIFY TESTS - ROC IS RETURNING A TUPLE. WE MAY ALSO DISCARD IT ####
-    # what is AUC metric telling at all? Computing it for prediction and ground truth
-    # is not making sense
-    # metrics that do not have any "average" parameter
-    # calculators = {
-    #
-    #     # "auc": tm.AUC(reorder=True),
-    #     ## weird error for multi-class problem, where pos_label is not getting set
-    #     "roc": tm.ROC(task=task, num_classes=params["model"]["num_classes"]),
-    # }
-    # for metric_name, calculator in calculators.items():
-    #     if metric_name == "roc":
-    #         one_hot_preds = one_hot(
-    #             prediction.long(), num_classes=params["model"]["num_classes"]
-    #         )
-    #         output_metrics[metric_name] = get_output_from_calculator(
-    #             one_hot_preds.float(), target, calculator
-    #         )
-    #     else:
-    #         output_metrics[metric_name] = get_output_from_calculator(
-    #             prediction, target, calculator
-    #         )
+    # Matthews correlation coefficient is only defined for classification
+    if params["problem_type"] == "classification":
+        output_metrics["mcc"] = matthews_corrcoef(
```

Comment:
Generally torchmetrics does provide MCC metric (https://lightning.ai/docs/torchmetrics/stable/classification/matthews_corr_coef.html#torchmetrics.classification.BinaryMatthewsCorrCoef), wouldn't it be more coherent to use this one as another calculator in the dict? Then we could also stay with computations on GPU and reduce the data copy overhead.

Reply:
You are right! I somehow missed that because I was informed to use the implementation from either MONAI or scikit-learn. Anyway, I have confirmed that all implementations give the same result for the unit test, so I will use torchmetrics. Thanks!